Weather: Add Hourly Forecasts.

akshatdalton opened the pull request:
Now, we show the next 4 hours forecasts
along with the current weather/temperature.

Fixes: #1.

As mentioned in the issue, I couldn't understand how to make time + date-based queries to the given API to fetch forecast information. I looked up their documentation and they don't support any such query parameters. If I am wrong then please do correct me on this.

Please review.
@Shikhar15606 
@esh04 
@rida-hash

khattakdev commented:
@akshatdalton 

send me the `OPEN_CAGE_APIKEY` in DM, so I can add it to netlify and we can test it.

Shikhar15606 reviewed (approved):
LGTM. Great Work @akshatdalton

khattakdev commented:
@akshatdalton (Forgot to mention this in the previous comment) Please update the `example.env` as well with `OPEN_CAGE_APIKEY`

khattakdev commented on src/Components/HourlyWeather/index.js:
```diff
@@ -0,0 +1,53 @@
+
+    useEffect(async () => { 
+        async function convertGeoLocation(placename) {
+            const response = await fetch(`https://api.opencagedata.com/geocode/v1/json?q=${placename}&key=${process.env.OPEN_CAGE_APIKEY}`).then(value => value.json());
+            const cord = response["results"][0]["geometry"];
+            return [cord["lat"], cord["lng"]];
```
Add Exceptional handling

In case the API call fails, it should show a nice error message.

Add this in a different PR but mentioning it here so you know it.

khattakdev commented on src/Components/HourlyWeather/index.js:
```diff
@@ -0,0 +1,53 @@
+const HourlyWeather = ({ placename }) => {
+    const [hourlyData, setHourlyData] = useState([]);
+
+    useEffect(async () => { 
+        async function convertGeoLocation(placename) {
+            const response = await fetch(`https://api.opencagedata.com/geocode/v1/json?q=${placename}&key=${process.env.OPEN_CAGE_APIKEY}`).then(value => value.json());
+            const cord = response["results"][0]["geometry"];
```
Make the useEffect function synchronous. This is a bad practice.

Make it something like this:

```
 async function apiCallsFn() {
    // Your Code

  }
  
  
  useEffect( () => {
    
    apiCallsFn();
  }, [placename]);
 ```

khattakdev commented on src/Components/HourlyWeather/index.js:
```diff
@@ -0,0 +1,53 @@
+
+    useEffect(async () => { 
+        async function convertGeoLocation(placename) {
+            const response = await fetch(`https://api.opencagedata.com/geocode/v1/json?q=${placename}&key=${process.env.OPEN_CAGE_APIKEY}`).then(value => value.json());
```
Any env variable should start with `REACT_API_`

akshatdalton commented:
Following changes done in current push:

- Added `REACT_API_OPEN_CAGE_APIKEY` in `example.env` file (https://github.com/MLH-Fellowship/prep-project-4.1.5/pull/11#discussion_r753647162, https://github.com/MLH-Fellowship/prep-project-4.1.5/pull/11#issuecomment-974272213)
- Made the `useEffect` function synchronous and added error handling which goes along with it. (https://github.com/MLH-Fellowship/prep-project-4.1.5/pull/11#discussion_r753395679, https://github.com/MLH-Fellowship/prep-project-4.1.5/pull/11#discussion_r753646844)

Also, currently, there are two directories with the same name: `components` and `Components` we should resolve this and should use which follows the convention.

@khattakdev, please review.

P.S.: I just resolved the conflicts and can see that Hourly Weather is shown in the bottom - will change this in the next PR.

khattakdev commented:
Dope 🔥
